File: src/my_py_pkg/my_py_pkg/my_first_node.py
#from tokenize import Name
import rclpy
from rclpy.node import Node

class MyNode(Node):
    def __init__(self):
        super().__init__("py_test")
        self.get_logger().info("Hi Aj")

        self.counter1 = 0
        self.create_timer(0.5,self.timer_callback,)
    
    def timer_callback(self):
        self.counter1 += 1
        self.get_logger().info("Aj " + str(self.counter1))

def main(args=None):
    rclpy.init(args=args)
    node = MyNode()
    rclpy.spin(node)
    
    rclpy.shutdown()

if __name__ == "__main__" :
    main()   

File: src/my_cpp_pkg/src/my_first_node.cpp
#include "rclcpp/rclcpp.hpp"

class MyNode: public rclcpp::Node {
public:
    MyNode(): Node("cpp_test"), counter_(0)
    {
    RCLCPP_INFO(this->get_logger(),"hello");

    timer_ = this->create_wall_timer(std::chrono::seconds(1),std::bind(&MyNode::timerCallback, this));
    }

private:
    void timerCallback() {
        counter_++;
        RCLCPP_INFO(this->get_logger(),"Hello Aj Node %d",counter_);
    }
    rclcpp::TimerBase::SharedPtr timer_;
    int counter_;
};

int main(int argc, char **argv) {
    rclcpp::init(argc, argv);
    auto node = std::make_shared<MyNode>();
    //RCLCPP_INFO(node->get_logger(),"Hi ");
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}  